File: app/src/main/java/com/zerostress/manager/models/Announcement.java
package com.zerostress.manager.models;

public class Announcement {
    private String id;
    private String text;
    private String author;
    private long timestamp;

    public Announcement() {}

    public Announcement(String text, String author) {
        this.text = text;
        this.author = author;
        this.timestamp = System.currentTimeMillis();
    }

    public String getId() { return id; }
    public String getText() { return text; }
    public String getAuthor() { return author; }
    public long getTimestamp() { return timestamp; }

    public void setId(String id) { this.id = id; }
    public void setText(String text) { this.text = text; }
    public void setAuthor(String author) { this.author = author; }
    public void setTimestamp(long timestamp) { this.timestamp = timestamp; }
}

File: app/src/main/java/com/zerostress/manager/models/Announcement.kt
package com.zerostress.manager.models

import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class Announcement(
    var id: String = "",
    var text: String = "",
    var author: String = "",
    var timestamp: Long = 0L
) {
    companion object {
        fun create(text: String, author: String): Announcement =
            Announcement(text = text, author = author, timestamp = System.currentTimeMillis())
    }
}
